flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -94,29 +108,91 @@ public class AsyncExecutionController<K> {
      */
     final AtomicInteger inFlightRecordNum;
 
+    /**
+     * The flag to indicate whether the {@link #bufferTimeOut} is reached, if yes, a trigger will
+     * perform actively when the next state request arrives even if the activeQueue has not reached
+     * the {@link #batchSize}.
+     */
+    final AtomicBoolean timeoutFlag;
+
+    /** The executor service that schedules and calls the triggers of this task. */
+    final ScheduledThreadPoolExecutor scheduledExecutor;
```

Comment:
Should AEC become `Closeable` since we maintain some resources like this internally ?

Reply:
👍 I added a `close()` method, 2 things include:
1. drain all in-flight records
2. shutdown `scheduledExecutor`

Reply:
Would it be better to reuse existing utility methods like `FutureUtils.delay()`? This way AEC won't need to maintain such resources.